ql/src/java/org/apache/hadoop/hive/ql/metadata/HiveStorageHandler.java

```diff
@@ -692,4 +692,9 @@ default void validatePartSpec(org.apache.hadoop.hive.ql.metadata.Table hmsTable,
     throw new UnsupportedOperationException("Storage handler does not support validation of partition values");
   }
 
+  default boolean shouldTruncate(org.apache.hadoop.hive.ql.metadata.Table hmsTable, Map<String, String> partitionSpec)
+      throws SemanticException {
+    return true;
```

Comment:
why if true by default?

Reply:
Other storage handlers can still perform truncate, hence no need to fallback to delete for other storage handlers.